### PR TITLE
Fix mlflow.keras.save_model with DBFS

### PR DESCRIPTION
**Pull request by andychow-db**
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixed a bug where `mlflow.keras.save_model` failed against "/dbfs/..." paths due to the lack of support for random writes when using DBFS FUSE.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [x] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by tomasatdatabricks — APPROVED** [review]
lgtm! thx @andychow-db

**Comment by tomasatdatabricks** [comment]
Hey @andychow-db is this ready for review / merge? I have already approved it, do you think it misses something? I would merge it otherwise.

File: mlflow/keras.py [hunk 1]
```diff
@@ -150,33 +152,58 @@ def _is_tf_keras(model):
     elif type(keras_module) == str:
         keras_module = importlib.import_module(keras_module)
 
+    # check if path exists
```
Comment (smurching) [inline mlflow/keras.py]: One small note on comments, we tend to prefer not adding them in favor of writing code that's easy to follow, reserving their use for cases where it's hard to tell exactly what's going on (e.g. here from the variable names & logic we can see that we're checking whether the path already exists). That's super fuzzy I know but hopefully it makes some sense - the idea is that comments can go out of date, so it's always better if the code is simple enough to follow on its own, but for more complex cases comments are definitely helpful

File: mlflow/keras.py [hunk 2]
```diff
@@ -150,33 +152,58 @@ def _is_tf_keras(model):
...
+    # save keras module spec to path/data/keras_module.txt
     with open(os.path.join(data_path, _KERAS_MODULE_SPEC_PATH), "w") as f:
         f.write(keras_module.__name__)
+
+    # save keras model to path/data/model.h5
     model_subpath = os.path.join(data_subpath, _MODEL_SAVE_PATH)
-    keras_model.save(os.path.join(path, model_subpath), **kwargs)
+    model_path = os.path.join(path, model_subpath)
+    if path.startswith('/dbfs/'):
+        # The Databricks Filesystem uses a FUSE implementation that does not support
```
Comment (smurching) [inline mlflow/keras.py]: This is a great example of a good comment, it's not obvious why we're checking for "/dbfs/" paths without it

**Review by smurching — APPROVED** [review]
LGTM! Just left a note on comments for the future

File: mlflow/keras.py (hunk 1 above)
Comment (andychow-db) [reply mlflow/keras.py]: Sounds good. I'll be sure to avoid superfluous comments in the future.

**Pull request merged**